core/src/main/kotlin/org/opensearch/alerting/core/lock/LockService.kt

```diff
@@ -72,13 +75,23 @@ class LockService(private val client: Client, private val clusterService: Cluste
                                 object : ActionListener<LockModel> {
                                     override fun onResponse(existingLock: LockModel?) {
                                         if (existingLock != null) {
+                                            val currentTimestamp = getNow()
                                             if (isLockReleased(existingLock)) {
                                                 log.debug("lock is released or expired: {}", existingLock)
                                                 val updateLock = LockModel(existingLock, getNow(), false)
                                                 updateLock(updateLock, listener)
                                             } else {
-                                                log.debug("Lock is NOT released or expired. {}", existingLock)
-                                                listener.onResponse(null)
+                                                log.debug("Lock is NOT released. {}", existingLock)
+                                                if (existingLock.lockTime.epochSecond + LOCK_EXPIRED_SECONDS.seconds
+                                                    < currentTimestamp.epochSecond
+                                                ) {
+                                                    log.debug("Lock is expired. Renewing Lock {}", existingLock)
+                                                    val updateLock = LockModel(existingLock, getNow(), false)
+                                                    updateLock(updateLock, listener)
+                                                } else {
+                                                    log.debug("Lock is NOT expired. {}", existingLock)
```

Comment:
plz explicitly mention we don't run `Lock is NOT expired. Not running monitor`

Reply:
fixed it.